src/hardhat/hre/W3fHardhatPlugin.ts

```diff
@@ -0,0 +1,119 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { Web3FunctionContextData, Web3FunctionUserArgs } from "../../lib";
+import { Web3FunctionBuilder } from "../../lib/builder";
+import { Web3FunctionExecSuccess, Web3FunctionRunner } from "../../lib/runtime";
+import { MAX_RPC_LIMIT } from "../constants";
+import { EthersProviderWrapper } from "../provider";
+import { W3fDetails } from "../types";
+import { getW3fDetails } from "../utils";
+
+export class W3fHardhatPlugin {
+  private hre: HardhatRuntimeEnvironment;
+
+  constructor(_hre: HardhatRuntimeEnvironment) {
+    this.hre = _hre;
+  }
+
+  public get(_name: string) {
+    const w3f = getW3fDetails(_name, this.hre.config.w3f.rootDir);
```

Comment:
I think it will be good to keep an in memory cache for this
So that when you run multiple tests, it can avoid reading all files everytime and just the first time instead

Reply:
If the user does not instantiate a new class every time, it should be fine right? Something like this https://github.com/gelatodigital/web3-functions-hardhat-template/blob/9bf932861af7feaeb5e9a7bfe16cc1609eda71e9/test/examples/CoingeckoOracle.test.ts#L30

Reply:
Hum yes ok, you're right